### PR TITLE
add(gpu):Add renesas-ra6m3 gpu adaptation

### PR DESCRIPTION
**Pull request by Rbb666**
### Description of the feature or fix

A clear and concise description of what the bug or new feature is.

### Checkpoints
- [x] Run `code-format.py` from the scripts folder. [astyle](http://astyle.sourceforge.net/install.html) needs to be installed.
- [ ] Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed
- [ ] Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant. 
- [ ] Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- [x] If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/release/v8.3/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/release/v8.3/Kconfig).
 
Be sure the following conventions are followed:
- [ ] Follow the [Styling guide](https://github.com/lvgl/lvgl/blob/master/docs/CODING_STYLE.md)
- [ ] Prefer `enum`s instead of macros. If inevitable to use `define`s export them with `LV_EXPORT_CONST_INT(defined_value)` right after the `define`.
- [ ] In function arguments prefer `type name[]` declaration for array parameters instead of `type * name`
- [ ] Use typed pointers instead of `void *` pointers
- [ ] Do not `malloc` into a static or global variables. Instead declare the variable in `LV_ITERATE_ROOTS` list in [`lv_gc.h`](https://github.com/lvgl/lvgl/blob/master/src/misc/lv_gc.h) and mark the variable with `GC_ROOT(variable)` when it's used. See a detaild description [here](https://docs.lvgl.io/master/get-started/bindings/micropython.html#memory-management).
- [ ] Widget constructor must follow the `lv_<widget_name>_create(lv_obj_t * parent)` pattern.
- [ ] Widget members function must start with `lv_<modul_name>` and should receive `lv_obj_t *` as first argument which is a pointer to widget object itself.  
- [ ] `struct`s should be used via an API and not modified directly via their elements.
- [ ] `struct` APIs should follow the widgets' conventions. That is to receive a pointer to the `struct` as the first argument, and the prefix of the `struct` name should be used as the prefix of the function name too (e.g.  `lv_disp_set_default(lv_disp_t * disp)`)
- [ ] Functions and `struct`s which are not part of the public API must begin with underscore in order to mark them as "private".
- [ ] Arguments must be named in H files too.
- [ ] To register and use callbacks one of the followings needs to be followed (see a detaild description [here](https://docs.lvgl.io/master/get-started/bindings/micropython.html#callbacks)): 
  - For both the registration function and the callback pass a pointer to a `struct` as the first argument. The `struct` must contain `void * user_data` field.
  - The last argument of the registration function must be `void * user_data` and the same `user_data` needs to be passed as the last argument of the callback.
  - Callback types not following these conventions should end with `xcb_t`.

### CODE REVIEW

**Comment by kisvegabor** [comment]
Hi,

Thank you for this PR. 

From v9 I'll handle GPUs in a special way. I started to establish a partnership with chip vendors to integrate their GPU support and I would like mainly the vendors to contribute and maintain their GPUs. 

Renesas was also open to it but we are still discussing the details. 

Besides that we can't add such a big update to v8, but it should be part of v9 and should work with the [new rendering architecture](https://github.com/lvgl/lvgl/issues/4016).

So in short, unfortunately I can't merge it now but I'm sure it will be a good starting point once we agree with Renesas abut the details of the partnership.

**Comment by kisvegabor** [comment]
I've discussed some details with RT-Thread and Renesas and I will merge the PR now to v8.

We will apply the "I would like mainly the vendors to contribute and maintain their GPUs" from v9.

Please send a new PR with some updates to the docs. Similar to [this](https://docs.lvgl.io/master/get-started/platforms/nxp.html).

**Comment by Rbb666** [comment]
> 我已经与RT-Thread和瑞萨电子讨论了一些细节，我现在将PR合并到v8。
> 
> 我们将应用 v9 中的“我希望主要供应商贡献和维护他们的 GPU”。
> 
> 请发送一个新的 PR，其中包含对文档的一些更新。[与此类似](https://docs.lvgl.io/master/get-started/platforms/nxp.html)。

Okay, we will provide the corresponding documents later.

**Comment by Rbb666** [comment]
> > 我已经与RT-Thread和瑞萨电子讨论了一些细节，我现在将PR合并到v8。
> > 我们将应用 v9 中的“我希望主要供应商贡献和维护他们的 GPU”。
> > 请发送一个新的 PR，其中包含对文档的一些更新。[与此类似](https://docs.lvgl.io/master/get-started/platforms/nxp.html)。
> 
> Okay, we will provide the corresponding documents later.

PR has been submitted, here is the [link](https://github.com/lvgl/lvgl/pull/4278).

File: src/draw/renesas/lv_gpu_d2_ra6m3.c [hunk 1]
```diff
@@ -0,0 +1,742 @@
...
+            break;
+        case LV_IMG_CF_RGBA8888:
+            d2_cf = d2_mode_rgba8888;
+            break;
+#endif /* DLG_LVGL_CF */
+        default:
+            return -1;
+    }
+
+#if (DLG_LVGL_CF == 1)
```
Comment (XuNeo) [inline src/draw/renesas/lv_gpu_d2_ra6m3.c:125]: @kisvegabor @FASTSHIFT 

It looks like lvgl needs a common way to support this requirement. Image format should be a combination of basic cf, compression methods(RLE for instance), buffer mode(linear or tiled, which `vglite` supports), raster or vector image.

Is this discussed already in v9?

Comment (kisvegabor) [reply src/draw/renesas/lv_gpu_d2_ra6m3.c:125]: Please ask this question in https://github.com/lvgl/lvgl/issues/4016
